data/tools/wesnoth/campaignserver_client.py

```diff
@@ -111,6 +111,16 @@ def __del__(self):
 
 
     def make_packet(self, doc):
+
+        # The Wesnoth campaign server only accepts "simple WML", which
+        # requires all attributes to be sorted alphabetically.
+        def sorter(tag):
+            if isinstance(tag, wmlparser.TagNode):
+                tag.data.sort(key = lambda node: node.name)
+                for tag2 in tag.data:
+                    sorter(tag2)
+        sorter(doc)
```

Comment:
I wonder why the function was named "sorter" rather than "sort" (verb). Also, do not fully understand why the function body wasn't just in-lined.

P.S. Big thanks for fixing it! Will try out the patch soon.

Reply:
I've tried the new wesnoth_addon_manager, it works.
Thanks!